packages/mdc-select/index.ts

```diff
@@ -0,0 +1,596 @@
+        selectedItemEl.focus();
+      }
+    };
+    this.handleMenuClosed_ = () => {
+      // menuOpened_ is used to track the state of the menu opening or closing since the menu.open function
+      // will return false if the menu is still closing and this method listens to the closed event which
+      // occurs after the menu is already closed.
+      this.menuOpened_ = false;
+      this.selectedText_!.removeAttribute('aria-expanded');
+      if (document.activeElement !== this.selectedText_) {
+        this.foundation_.handleBlur();
+      }
+    };
+
+    // One of these elements must be non-null.
+    const element = this.nativeControl_ || this.selectedText_!;
+
+    element.addEventListener('change', this.handleChange_);
+    element.addEventListener('focus', this.handleFocus_);
+    element.addEventListener('blur', this.handleBlur_);
+
+    MOUSEDOWN_TOUCHSTART_EVENTS.forEach((evtType) => {
+      element.addEventListener(evtType, this.handleClick_ as EventListener);
+    });
+
+    if (this.menuElement_) {
+      this.selectedText_!.addEventListener('keydown', this.handleKeydown_);
+      this.menu_!.listen(menuSurfaceConstants.strings.CLOSED_EVENT, this.handleMenuClosed_);
+      this.menu_!.listen(menuSurfaceConstants.strings.OPENED_EVENT, this.handleMenuOpened_);
+      this.menu_!.listen(menuConstants.strings.SELECTED_EVENT, this.handleMenuSelected_);
+
+      if (this.hiddenInput_ && this.hiddenInput_.value) {
+        // If the hidden input already has a value, use it to restore the select's value.
+        // This can happen e.g. if the user goes back or (in some browsers) refreshes the page.
+        const enhancedAdapterMethods = this.getEnhancedSelectAdapterMethods_();
+        enhancedAdapterMethods.setValue(this.hiddenInput_.value);
+      } else if (this.menuElement_.querySelector(strings.SELECTED_ITEM_SELECTOR)) {
+        // If an element is selected, the select should set the initial selected text.
+        const enhancedAdapterMethods = this.getEnhancedSelectAdapterMethods_();
+        enhancedAdapterMethods.setValue(enhancedAdapterMethods.getValue());
+      }
+    }
+
+    // Initially sync floating label
+    this.foundation_.handleChange(/* didChange */ false);
+
+    if (this.root_.classList.contains(cssClasses.DISABLED)
+      || (this.nativeControl_ && this.nativeControl_.disabled)) {
+      this.disabled = true;
+    }
+  }
+
+  destroy() {
+    // One of these elements must be non-null.
+    const element = this.nativeControl_ || this.selectedText_!;
+
+    element.removeEventListener('change', this.handleChange_);
+    element.removeEventListener('focus', this.handleFocus_);
+    element.removeEventListener('blur', this.handleBlur_);
+    element.removeEventListener('keydown', this.handleKeydown_);
+    MOUSEDOWN_TOUCHSTART_EVENTS.forEach((evtType) => {
+      element.removeEventListener(evtType, this.handleClick_ as EventListener);
```

Comment:
same with here. I think we need a guard, since this.selectedText_ can be null

Reply:
`element` should never be null:

```ts
// One of these elements must be non-null.
const element = this.nativeControl_ || this.selectedText_!;
```